src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -234,6 +238,54 @@ public function getYamlCompileTests()
             $container,
         ];
     }
+
+    public function testTaggedServiceWithIndexAttribute()
+    {
+        $container = new ContainerBuilder();
+        $container->register('bar_tag_service', BarTagClass::class)
```

Comment:
you can use service id == class name for all these services. this should make it easier.

Reply:
yes indeed